internal/jasmine_node_test/jasmine_runner.js

```diff
@@ -11,45 +12,94 @@ const UTF8 = {
+const IBAZEL_NOTIFY_BUILD_SUCCESS = 'IBAZEL_BUILD_COMPLETED SUCCESS';
+
 // Set the StackTraceLimit to infinity. This will make stack capturing slower, but more useful.
 // Since we are running tests having proper stack traces is very useful and should be always set to
 // the maximum (See: https://nodejs.org/api/errors.html#errors_error_stacktracelimit)
 Error.stackTraceLimit = Infinity;
 
-function main(args) {
-  if (!args.length) {
-    throw new Error('Spec file manifest expected argument missing');
-  }
-  const manifest = require.resolve(args[0]);
-  // Remove the manifest, some tested code may process the argv.
-  process.argv.splice(2, 1)[0];
 
-  const jrunner = new JasmineRunner();
+function watchMode(manifestLocation) {
+  // ibazel communicates with us via stdin
+  const rl = createInterface({input: process.stdin, terminal: false});
+  rl.on('line', chunk => {
+    console.log('new line')
+    if (chunk === IBAZEL_NOTIFY_BUILD_SUCCESS) {
+      //TODO: we should try keep the same instance of JasmineRunner if it's expensive to new up
+      //TODO: stop previous runs if we get multiple calls of this in quick succession
+      const jrunner = createRunner(true);
+      addFiles(manifestLocation, jrunner);
+      jrunner.execute();
+    }
+  });
+  rl.on('close', () => {
+    console.log('someone trying to exit me')
+    // Give ibazel 5s to kill our process, otherwise do it ourselves
+    setTimeout(() => {
+      console.error('ibazel failed to stop jasmine after 5s; probably a bug');
+      process.exit(1);
+    }, 5000);
+  });
+}
+
+function addFiles(manifestLocation, jrunner){
+  const manifest = require.resolve(manifestLocation);  
+  console.log(manifest)
   fs.readFileSync(manifest, UTF8)
-      .split('\n')
-      .filter(l => l.length > 0)
-      .forEach(f => jrunner.addSpecFile(f));
+    .split('\n')
+    .filter(l => l.length > 0)
+    .forEach(f => jrunner.addSpecFile(f));
+}
+
+function createRunner(watch) {
+  const jrunner = new JasmineRunner();
 
   var noSpecsFound = true;
   jrunner.addReporter({
     specDone: () => {
       noSpecsFound = false
     },
   });
+  
   // addReporter throws away the default console reporter
   // so we need to add it back
   jrunner.configureDefaultReporter({});
 
   jrunner.onComplete((passed) => {
     let exitCode = passed ? 0 : BAZEL_EXIT_TESTS_FAILED;
     if (noSpecsFound) exitCode = BAZEL_EXIT_NO_TESTS_FOUND;
-    process.exit(exitCode);
+    !watch ? process.exit(exitCode) : null;
   });
+  return jrunner;
+}
+
+function main(args) {
+  if (!args.length) {
+    throw new Error('Spec file manifest expected argument missing');
+  }
+  const manifestLocation = args[0];
+  // Remove the manifest, some tested code may process the argv.
+  process.argv.splice(2, 1)[0];
+  
+  const watch = true;
+  const jrunner = createRunner(watch);
+  addFiles(manifestLocation, jrunner);
 
-  jrunner.execute();
-  return 0;
+  // TODO: find out how to determine if we're running under ibazel
```

Comment:
use the environment variable
https://github.com/bazelbuild/bazel-watcher/pull/60/files#diff-79db8648ad38b76a407bbc7744e0a7aeR74